### PR TITLE
Add `check_input` option to `UnitaryGate`

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a new option to the unitary gate class's constructor, skip_check, which is used to skip the checking on the input unitary. When creating a unitary gates when you know the input is already unitary checking that the input matrix is unitary is wasted CPU time. This new option enables a user to assert the input is unitary and skip this checking (at their own risk). The primary use case for this is the consolidate blocks pass which is creating a large number of UnitaryGate objects and matrix is always going to be unitary. This new flag is used in ConsolidateBlocks to speed up the creation of gate objects.

### Details and comments

### CODE REVIEW

File: qiskit/extensions/unitary.py [hunk 1]
```diff
@@ -56,12 +56,17 @@ class UnitaryGate(Gate):
             circuit.append(gate, [0, 1])
     """
 
-    def __init__(self, data, label=None):
+    def __init__(self, data, label=None, skip_check=False):
```
Comment (Cryoris) [inline qiskit/extensions/unitary.py:59]: That's a good option to have 👍🏻 We have a similar option in the SK passes, where it's called `check_input` -- do we want to call it the same here for consistency?

Comment (mtreinish) [reply qiskit/extensions/unitary.py:59]: Sure, I'll rename the option.

Comment (mtreinish) [reply qiskit/extensions/unitary.py]: Renamed in: https://github.com/Qiskit/qiskit/pull/10660/commits/387088561c38c0537574e90007c18046afcf6335

File: qiskit/circuit/library/generalized_gates/unitary.py [hunk 2]
```diff
@@ -63,13 +63,24 @@ class UnitaryGate(Gate):
...
             data: Unitary operator.
             label: Unitary name for backend [Default: None].
+            check_input: If set to ``False`` this asserts the input
+                is known to be unitary and the checking to validate this will
+                be skipped. This should only ever be used if you know the
+                input is unitary, setting this to ``False`` and passing in
+                a non-unitary matrix will result unexpected behavior and errors.
 
-        Raises:
+        Raises
```
Comment (jakelishman) [inline qiskit/circuit/library/generalized_gates/unitary.py]: I think this colon got lost?

**Review by jakelishman — COMMENTED** [review]
This looks easy and sensible - only point is that I think a colon got lost by accident in a docstring.

File: qiskit/circuit/library/generalized_gates/unitary.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/circuit/library/generalized_gates/unitary.py]: Oops yep

Comment (mtreinish) [inline qiskit/circuit/library/generalized_gates/unitary.py]:
```suggestion
        Raises:
```

**Review by jakelishman — APPROVED** [review]

**Pull request merged**